File: Module.Persist.TPM/Model/TPM/PromoSupport.cs
using Core.Data;
using Module.Persist.TPM.Model.Interfaces;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;

namespace Module.Persist.TPM.Model.TPM
{
    public class PromoSupport : IEntity<Guid>, IDeactivatable
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public System.Guid Id { get; set; }
        public bool Disabled { get; set; }
        public DateTimeOffset? DeletedDate { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTimeOffset? ModifiedDate { get; set; }

        public int Number { get; set; }
        public int? PlanQuantity { get; set; }
        public int? ActualQuantity { get; set; }
        public double? PlanCostTE { get; set; }
        public double? ActualCostTE { get; set; }
        public DateTimeOffset? StartDate { get; set; }
        public DateTimeOffset? EndDate { get; set; }
        public double? PlanProdCostPer1Item { get; set; }
        public double? ActualProdCostPer1Item { get; set; }
        public double? PlanProdCost { get; set; }
        public double? ActualProdCost { get; set; }
        public string UserTimestamp { get; set; }
        public string AttachFileName { get; set; }
        public string BorderColor { get; set; }
        public string PONumber { get; set; }
        public string InvoiceNumber { get; set; }
        public bool OffAllocation { get; set; }

        public int ClientTreeId { get; set; }
        public virtual ClientTree ClientTree { get; set; }
        public Guid? BudgetSubItemId { get; set; }
        public virtual BudgetSubItem BudgetSubItem { get; set; }

        public ICollection<PromoSupportPromo> PromoSupportPromo { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/promosupport/PromoSupport.js
Ext.define('App.model.tpm.promosupport.PromoSupport', {
    extend: 'Ext.data.Model',
    idProperty: 'Id',
    breezeEntityType: 'PromoSupport',
    fields: [
        { name: 'Id', hidden: true },
        { name: 'ClientTreeId', hidden: true, isDefault: true },
        { name: 'BudgetSubItemId', hidden: true, isDefault: true },
        { name: 'PromoId', useNull: true, hidden: true, isDefault: false, defaultValue: null },
        { name: 'Number', type: 'int', hidden: false, isDefault: true },
        { name: 'PlanQuantity', type: 'int', hidden: false, isDefault: true },
        { name: 'ActualQuantity', type: 'int', hidden: false, isDefault: true },
        { name: 'PlanCostTE', type: 'float', hidden: false, isDefault: true },
        { name: 'ActualCostTE', type: 'float', hidden: false, isDefault: true },
        { name: 'StartDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'EndDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'PlanProdCost', type: 'float', hidden: true, isDefault: false },
        { name: 'ActualProdCost', type: 'float', hidden: true, isDefault: false },
        { name: 'UserTimestamp', type: 'string', hidden: true, isDefault: false },
        { name: 'AttachFileName', type: 'string', hidden: true, isDefault: false },
        { name: 'BorderColor', type: 'string', hidden: true, isDefault: false },
        { name: 'PlanProdCostPer1Item', type: 'float', hidden: false, isDefault: true },
        { name: 'ActualProdCostPer1Item', type: 'float', hidden: false, isDefault: true },
        { name: 'PONumber', type: 'string', hidden: false, isDefault: true },
        { name: 'InvoiceNumber', type: 'string', hidden: false, isDefault: true },
        {
            name: 'ClientTreeFullPathName', type: 'string', mapping: 'ClientTree.FullPathName', tree: true,
            defaultFilterConfig: { valueField: 'FullPathName' }, breezeEntityType: 'ClientTree', hidden: false, isDefault: true
        },
        {
            name: 'BudgetSubItemName', type: 'string', mapping: 'BudgetSubItem.Name', defaultFilterConfig: { valueField: 'Name' },
            breezeEntityType: 'BudgetSubItem', hidden: false, isDefault: true
        },
        {
            name: 'BudgetSubItemBudgetItemName', type: 'string', mapping: 'BudgetSubItem.BudgetItem.Name', defaultFilterConfig: { valueField: 'Name' },
            breezeEntityType: 'BudgetItem', hidden: false, isDefault: true
        },
        {
            name: 'BudgetSubItemBudgetItemId', mapping: 'BudgetSubItem.BudgetItem.Id', defaultFilterConfig: { valueField: 'Id' },
            breezeEntityType: 'BudgetItem', hidden: true, isDefault: false
        }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'PromoSupports',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});
